lib/mjsonwp/mjsonwp.js

```diff
@@ -324,19 +324,17 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
       // if anything goes wrong, figure out what our response should be
       // based on the type of error that we encountered
       let actualErr = err;
-      if (driver.protocol === BaseDriver.DRIVER_PROTOCOL.MJSONWP) {
-        if (isErrorType(err, errors.ProxyRequestError)) {
-          log.error(`Encountered internal error running command:  ${JSON.stringify(err)} ${err.stack}`);
-          actualErr = err.getActualError();
-        } else if (!(isErrorType(err, MJSONWPError) ||
-          isErrorType(err, errors.BadParametersError))) {
-          log.error(`Encountered internal error running command: ${err.stack}`);
-          actualErr = new errors.UnknownError(err);
-        }
-        [httpStatus, httpResBody] = getResponseForJsonwpError(actualErr);
-      } else {
-        [httpStatus, httpResBody] = getResponseForW3CError(actualErr);
+
+      // If it's a JWProxy error, translate it to a MJSONWPError exception
+      if (isErrorType(err, errors.ProxyRequestError)) {
+        log.error(`Encountered internal error running command:  ${JSON.stringify(err)} ${err.stack}`);
+        actualErr = err.getActualError(driver.protocol);
+      } else if (driver.protocol === BaseDriver.DRIVER_PROTOCOL.MJSONWP && (!(isErrorType(err, MJSONWPError) || isErrorType(err, errors.BadParametersError)))) {
```

Comment:
`driver.protocol === BaseDriver.DRIVER_PROTOCOL.MJSONWP` is pretty unwieldy, and is used in a number of places. Could it be extracted into helper functions (`driver.isMjsonwpProtocol()` and `driver.isW3CProtocol()` or something)? Also limits the number of places that know about BaseDriver's internals